src/gui/editorwidgets/qgsdatetimeedit.cpp

```diff
@@ -73,6 +89,7 @@ void QgsDateTimeEdit::setEmpty()
   mNullLabel->setVisible( false );
   lineEdit()->setVisible( false );
   mClearButton->setVisible( mAllowNull );
+  mIsEmpty = true;
```

Comment:
@nyalldawson following my comment in 6d868598f5e477c4dfe25a0acd80e6370e291f2b 
is this correct?

Reply:
Yes - thanks!